components/suggest/src/rs.rs

```diff
@@ -69,17 +83,58 @@ pub(crate) trait Client {
     fn get_records(&self, request: RecordRequest) -> Result<Vec<Record>>;
 }
 
-impl Client for remote_settings::Client {
+/// Implements the [Client] trait using a real remote settings client
+pub struct RemoteSettingsClient {
```

Comment:
Nit: maybe this should be plural as it has two clients now?

Reply:
I kind of like keeping it singular, since from the store's perspective it's a single client that's used to fetch data from remote settings.  Maybe this is just laziness though, since if I changed the name of the type, I'd also want to change the name of the field, maybe the name of the trait, etc.  I'm not that tied to this opinion though, happy to discuss it more in another issue/PR.

Reply:
It's really a nitpick, singular is also fine.